src/Illuminate/Queue/RedisQueue.php

```diff
@@ -64,7 +64,11 @@ public function __construct(Database $redis, $default = 'default',
      */
     public function size($queue = null)
     {
-        return (int) $this->getConnection()->llen($this->getQueue($queue));
+        $queue = $this->getQueue($queue);
+
+        return $this->getConnection()->llen($queue)
+            + $this->getConnection()->llen($queue.':delayed')
```

Comment:
This will throw an exception i guess. delayed queue is not a list and does not support llen. zcard should be used instead.